tiledb/sm/filter/webp_filter.h

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file   webp_filter.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class WebpFilter.
+ */
+
+#ifndef TILEDB_WEBP_FILTER_H
+#define TILEDB_WEBP_FILTER_H
+
+#ifdef TILEDB_WEBP
+
+#include "tiledb/common/status.h"
+#include "tiledb/sm/enums/filter_option.h"
+#include "tiledb/sm/enums/filter_type.h"
+#include "tiledb/sm/filter/filter.h"
+#include "tiledb/sm/misc/types.h"
+
+using namespace tiledb::common;
+
+namespace tiledb::sm {
+
+enum class WebpInputFormat : uint8_t;
+
+/**
+ * The WebP filter provides three options: quality, format, and lossless
+ *
+ * The quality option is used as quality_factor setting for WebP lossy
+ * compression and expects a float value in the range of 0.0f - 100.0f
+ * Quality of 0 corresponds to low quality and small output sizes, whereas 100
```

Comment:
A quality value of 0 represents the lowest quality (image?) and the smallest output size, whereas a quality value of 100 represents the highest quality (image) and largest output size.